core/src/main/java/org/infinispan/factories/ExpirationManagerFactory.java

```diff
@@ -13,13 +14,16 @@
  * @author William Burns
  * @since 8.0
  */
-@DefaultFactoryFor(classes = ExpirationManager.class)
+@DefaultFactoryFor(classes = {ExpirationManager.class, InternalExpirationManager.class})
 public class ExpirationManagerFactory extends AbstractNamedCacheComponentFactory implements
          AutoInstantiableFactory {
 
    @Override
    @SuppressWarnings("unchecked")
    public <T> T construct(Class<T> componentType) {
+      if (componentType.equals(ExpirationManager.class)) {
+         return componentType.cast(componentRegistry.getOrCreateComponent(InternalExpirationManager.class));
```

Comment:
IMO any component that injects an `ExpirationManager` should inject an `InternalExpirationManager` instead.

Reply:
Yeah I kept going back and forth on this one. Originally I had it throw an exception to tell the user/component that they have to Inject the `InternalExpirationManager`. But maybe we should just have it not have any component factory for `ExpirationManager`.